src/network/networkd-address-pool.c

```diff
@@ -131,7 +134,19 @@ int address_pool_acquire(AddressPool *p, unsigned prefixlen, union in_addr_union
                 return 0;
 
         u = p->in_addr;
-        for (;;) {
+
+        for (i = 0; p->family == AF_INET6 ? i < RANDOM_PREFIX_TRIAL_MAX : true; i++) {
+                if (p->family == AF_INET6) {
+                        _cleanup_free_ char *buf = NULL;
+
+                        r = in_addr_random_prefix(p->family, &u, p->prefixlen, prefixlen);
+                        if (r <= 0)
+                                return r;
+
+                        (void) in_addr_to_string(p->family, &u, &buf);
+                        log_debug("Address pool: Address %s/%u is generated.", strnull(buf), prefixlen);
```

Comment:
enclose these two lines and the _cleanup_free line above in `if (DEBUG_LOGGING)`?